marketplace_standard_app_api/main.py

```diff
@@ -108,29 +108,155 @@ async def heartbeat() -> HTMLResponse:
         500: {"description": "Internal server error."},
         501: {"description": "Not implemented."},
         503: {"description": "Service unavailable."},
     },
+    description="Create a dataset.\n" + CREATE_DATASET_DESCRIPTION,
 )
-async def create_dataset(dataset: DatasetModel) -> DatasetCreateResponse:
-    """Create a new dataset."""
+async def create_dataset(
+    file: UploadFile,
+    dataset_id: Optional[DatasetId] = None,
+) -> DatasetCreateResponse:
+    """Create a new or replace an existing dataset."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.post(
+    "/datasets/",
+    operation_id="createDatasetMetadata",
+    name="Create Dataset Metadata",
+    tags=["DataSink"],
+    status_code=202,
+    responses={
+        401: {"description": "Not authenticated."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+@api.post(
+    "datasets/{dataset_id}",
+    operation_id="createOrReplaceDatasetMetadata",
+    name="Create or Replace Dataset Metadata",
+    tags=["DataSink"],
+    status_code=202,
+    responses={
+        401: {"description": "Not authenticated."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def create_or_replace_dataset_metadata(
+    dataset_id: Optional[DatasetId] = None,
+):
+    """Create or replace dataset metadata.
+
+    Note: This operation is in compliance with the OpenStack Swift object
+    storage API:
+    https://docs.openstack.org/api-ref/object-store/index.html#create-or-update-object-metadata
+    """
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.head(
+    "/datasets/{dataset_id}",
+    operation_id="getDatasetMetadata",
+    tags=["DataSource"],
+    status_code=200,
+    response_class=Response,
+    responses={
+        401: {"description": "Not authenticated."},
+        404: {"description": "Not found."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def get_dataset_metadata(dataset_id: DatasetId) -> Response:
+    """Get dataset metadata.
+
+    Returns the dataset metadata in the response header in the form of:
+
+    - X-Object-Metadata-name: value
+
+    Where 'name' is the name of the metadata key and 'value' is the
+    corresponding value.
+
+    Example response header for a plain-text file:
+    - Content-Type: text/plain;charset=UTF-8
+    - Content-Length: 1234
+    - X-Object-Meta-my-key: some-value
+
+    Note: This operation is in compliance with the OpenStack Swift object
+    storage API:
+    https://docs.openstack.org/api-ref/object-store/index.htmld#show-object-metadata
```

Comment:
Broken link. It seems this was the intention: https://docs.openstack.org/api-ref/object-store/index.html#show-object-metadata

Reply:
```suggestion
    https://docs.openstack.org/api-ref/object-store/index.html#show-object-metadata
```